src/main/java/frc/robot/subsystems/Drive.java

```diff
@@ -172,35 +178,26 @@ public CommandBase driveAxis(double distance) {
         });
     }
 
-    double targetComponent = 0;
-    PIDController componentPID = new PIDController(0, 0, 0);
-
-    public CommandBase driveX(double distanceMeters) {
-        return cmd().onInitialize(() -> {
-            componentPID = drivePID.copyTranslationPidController();
-            componentPID.setTolerance(0.02);
-            targetComponent = pose.getY() + distanceMeters;
-            Logger.getInstance().recordOutput("targetX", targetComponent);
-        }).onExecute(() -> {
-            move(componentPID.calculate(pose.getY(), targetComponent), 0, 0);
-            Logger.getInstance().recordOutput("poseX", pose.getY());
-        }).runsUntil(componentPID::atSetpoint).onEnd(() -> move(0, 0, 0));
-    }
-
-    public CommandBase driveY(double distanceMeters) {
-        return cmd().onInitialize(() -> {
-            componentPID = drivePID.copyTranslationPidController();
-            componentPID.setTolerance(0.02);
-            targetComponent = pose.getX() + distanceMeters;
-        }).onExecute(() -> {
-            move(0, componentPID.calculate(pose.getX(), targetComponent), 0);
-        }).runsUntil(componentPID::atSetpoint).onEnd(() -> move(0, 0, 0));
+    private Pose2d relativeTarget = new Pose2d();
+
+    public CommandBase driveRelative(Translation2d translation, double timeoutSeconds) {
+        return sequence(
+                runOnce(() -> {
+                    relativeTarget = new Pose2d(
+                            pose.getX() + translation.getX(),
+                            pose.getY() + translation.getY(),
+                            pose.getRotation());
+                    drivePID.reset(pose.getTranslation());
+                }),
+                race(new FunctionalWaitCommand(() -> timeoutSeconds),
+                        driveTo(() -> relativeTarget, 0.01)));
     }
 
     private void deadbandMove(final double xSpeed, final double ySpeed,
             final double rotation, boolean isRobotCentric) {
 
-        var deadband = RobotUtils.scalingDeadband(0.15);
+        var deadband = RobotUtils.scalingDeadband(
+                (DriverStation.isFMSAttached()) ? 0.05 : 0.15);
```

Comment:
Out of curiosity, why is this deadband different when connected to the FMS?

Reply:
We were using a new controller without any joystick drift for the driver and we used the old one for testing on my laptop which had a bunch of drift, so we just had it so it would have a higher deadband when just testing.